acl/acl.go

```diff
@@ -307,6 +340,34 @@ func (a *ACL) matchingNamespaceCapabilitySet(ns string) (capabilitySet, bool) {
 	return a.findClosestMatchingGlob(a.wildcardNamespaces, ns)
 }
 
+// anyNamespaceAllows returns true if any namespace in the ACL object allows
+// the given operation.
+// If op is an empty string it returns true if any namespace allows any
+// operation.
```

Comment:
It feels like passing an empty op is a programmer error, not a condition we intentionally support. Should this return `ErrPermissionDenied` so that we can catch that in testing?

Reply:
Yeah, I didn't like this very much but figured it was OK given it's a private method. The key is that it needs to differentiate between two scenarios:

- `AllowNamespaceOperation` needs to know if any namespace allows a specific operation
- `AllowNamespace` needs to know if any namespace allows any operation (i.e., all don't deny)

Having a "magic" input (empty string in this case) serves as that switch between the two cases.

But looking at the code I found a better way to do it, so thanks for raising it!